cores/esp32/esp32-hal-log.c

```diff
@@ -0,0 +1,42 @@
+#ifndef __MY_LOG__
+#define __MY_LOG__
+#include "stdio.h"
+#include "esp32-hal-log.h"
+#ifdef USE_ESP32_LOG
+    void log_to_esp(esp_log_level_t level, const char *format, ...)
+    {   
+        va_list va_args;
+        va_start(va_args, format);
+
+        char log_buffer[512];
+        int len = vsnprintf(log_buffer, sizeof(log_buffer), format, va_args);
+        if (len > 0)
+        {
+            switch (level)
```

Comment:
this switch can be replaced by the following line:
```cpp
ESP_LOG_LEVEL_LOCAL(level, tag, "%s", log_buffer);
```

Reply:
It was my decision to doesn't call directly ESP_LOG_LEVEL_LOCAL, instead using the provided MACROs.
Bu i agree that the switch it's resolved in this way.

Reply:
by the way, the following should also work:
```cpp
ESP_LOG_LEVEL_LOCAL(level, tag, (const char *)log_buffer);
```

Reply:
No, in this way the compiler says :
` expression preceding parentheses of apparent call must have (pointer-to-) function typeC/C++(109)`

Reply:
ok :)